### ISSUE TITLE
Support nested params

### ISSUE THREAD

**Issue by igas** [palkan/rubanok#18]
## Is your feature request related to a problem? Please describe.

We have an API design where params are nested, like `filter[state]=paid`.

## Describe the solution you'd like

It would be great if map would accept hash/array as a path in params.

## Describe alternatives you've considered

- Changing API to flat structure
- Writing preprocessor

**Comment by palkan** [comment]
Wouldn't `match` cover this use case? Like in our examples:

```ruby
match :filter do
  having "active" do
    raw.active
  end

  having "finished" do
    raw.finished
  end
end
```

**Comment by igas** [comment]
Nope, as value is UGC. We'd like something like:
```ruby
map :filter do
  map :state do |state|
    raw.where(state:)
  end
end
```

**Comment by palkan** [comment]
I see.

I think, we can enhance `#match` to make it possible to use `#map` inside (since `#match` already has this nesting logic):

```ruby
match :filter do
  map :state do |state|
    # ...
  end
end
```

Will try.

**Comment by palkan** [comment]
I ended up with a new DSL method, `.process`:

```ruby
process :filter do
  map :state do |state|
    # ...
  end
end
```

Released in 0.5.0
